subsys/bluetooth/audio/mics_internal.h

```diff
@@ -9,11 +9,45 @@
 #include <zephyr/types.h>
 #include <bluetooth/gatt.h>
 
-int bt_mics_client_included_get(struct bt_conn *conn,
+struct bt_mics_server {
+	uint8_t mute;
+	struct bt_mics_cb *cb;
+	struct bt_gatt_service *service_p;
+	struct bt_aics *aics_insts[CONFIG_BT_MICS_AICS_INSTANCE_COUNT];
+};
+
+struct bt_mics_client {
+	uint16_t start_handle;
+	uint16_t end_handle;
+	uint16_t mute_handle;
+	struct bt_gatt_subscribe_params mute_sub_params;
+	struct bt_gatt_discover_params mute_sub_disc_params;
+
+	bool busy;
+	uint8_t mute_val_buf[1]; /* Mute value is a single octet */
+	struct bt_gatt_write_params write_params;
+	struct bt_gatt_read_params read_params;
+	struct bt_gatt_discover_params discover_params;
+	struct bt_conn *conn;
+
+	uint8_t aics_inst_cnt;
+	struct bt_aics *aics[CONFIG_BT_MICS_CLIENT_MAX_AICS_INST];
+};
+
+/* Struct used as a common type for the api */
+struct bt_mics {
```

Comment:
This construct would waste a lot of memory for the smallest of the server and client. I suggest to use the approach of bt_uuid to create a generic pointer than has a better memory footprint.